lib/Mage.js

```diff
@@ -3,22 +3,67 @@ var inherits = require('inherits');
 var EventManager = require('./EventManager');
 var CommandCenter = require('./CommandCenter');
 var MsgServer = require('./MsgServer');
+var HttpRequest = require('./HttpRequest');
 
 
-function Mage(config) {
+function Mage() {
 	EventEmitter.call(this);
 
 	this.eventManager = new EventManager();
 	this.msgServer = new MsgServer(this.eventManager);
 	this.commandCenter = new CommandCenter(this.eventManager);
-
-	this.configure(config);
 }
 
 inherits(Mage, EventEmitter);
 
 module.exports = Mage;
 
+// call config.get usercommand to get client config and init mage sdk with it
+//
+// the usercommand is called manually because CommandCenter needs the config to setup
+
+Mage.prototype.init = function (cb) {
+	if (!this.endpoint) {
+		throw new Error('Please call Mage.setEndpoint before calling Mage.init');
+	}
+
+	var that = this;
+	var hr = new HttpRequest({
```

Comment:
I think we discussed this, but can't we use `sendCommand` somehow? For the record, why couldn't we use it?